### PR TITLE
[AF-1555]: Upgrade WF11 to WF14

### PR DESCRIPTION
**Pull request by adrielparedes**
## Description

Upgrate to Wildfly 14

## Related to:
* https://github.com/jboss-integration/jboss-integration-platform-bom/pull/427
* https://github.com/errai/errai/pull/363
* https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/843
* https://github.com/kiegroup/appformer/pull/483
* https://github.com/kiegroup/kie-wb-common/pull/2147
* https://github.com/kiegroup/jbpm/pull/1336
* https://github.com/kiegroup/jbpm-wb/pull/1206
* https://github.com/kiegroup/drools/pull/2086
* https://github.com/kiegroup/droolsjbpm-integration/pull/1583
* https://github.com/kiegroup/kie-wb-distributions/pull/821 
* https://github.com/kiegroup/kie-docs/pull/1108
* https://github.com/kiegroup/optaplanner/pull/437
* https://github.com/kiegroup/optaplanner-wb/pull/304

### CODE REVIEW

**Comment by mswiderski** [comment]
@adrielparedes first of all changing the profile will essientilay disable any Jenkins verification as PR builds are executed with wildfly11 so if there is no such profile it will simply not execute any kie server integration tests.
Moreover you're missing main updates when it comes to upgrade of Jackson
https://github.com/mswiderski/droolsjbpm-integration/commit/7a22f22650c8a3d149e874d03647a560d9c55f17

**Comment by jhrcek** [comment]
@mswiderski 's first objections can probably be addressed by updating the profile names in  https://github.com/kiegroup/kie-jenkins-scripts/search?q=wildfly11

**Comment by mswiderski** [comment]
@jhrcek yes and by doing that will disable tests for all other PRs isn't it?

**Comment by jhrcek** [comment]
@mswiderski the way I understand it (I'd like to get this confirmed from @mbiarnes or @pszubiak) is that if you change the groovy jenkins DSL scripts on master, it will only affect jobs building from master. If you change those scripts on 7.11.x, it will affect only scripts building from 7.11.x branch etc.

For example couple of days ago I updated version of firefox for selenium tests on master and in 7.11.x jobs I still see old version is used (which is what I wanted!).

**Comment by pszubiak** [comment]
@jhrcek @mswiderski Yes it works this way. If we want to update Firefox version for 7.11.x we need to cherry pick [1] to 7.11.x branch.

[1]https://github.com/kiegroup/kie-jenkins-scripts/commit/38e922b6f183748ce1c4d6899dad007ae864d6d8

**Comment by jhrcek** [comment]
@pszubiak thanks, this confirms my theory. No, in this thread we're not discussion firefox (I just used it as recent example), but changes in this PR: renaming profiles `wildfly11` to `wildfly14`. What I wanted to confirm is that if we rename that profile in kie-jenkins-scripts on master, it will only affects jobs building from master branch and won't break older branches.

**Comment by mswiderski** [comment]
and that's exactly what I meant ... until this gets merged all other PRs for master won't run any tests as the profile won't match

**Comment by jhrcek** [comment]
@mswiderski oh right, didn't realise that. What can we do with that? the only thing that comes to my mind is split these changes into 2 steps:
1. update version but not profile names so we get full testing in PR jobs
Only after that works
2. update names of the profiles in all the PRs and in kie-jenkins-scipt (so we don't have "lying" configuration).

**Comment by jhrcek** [comment]
+1 although I wouldn't leave it for long term but change it after the tests are passing so we reduce our technical debt in this area (having version in property name seems bad enough to me to fix asap).

**Comment by adrielparedes** [comment]
@jhrcek @mswiderski Thanks for the review, I will upload missing changes, and rollback profiles to wildfly11

**Comment by MarianMacik** [comment]
jenkins retest this

**Comment by mswiderski** [comment]
@adrielparedes this commit is actually also needed 
https://github.com/kiegroup/droolsjbpm-integration/commit/2aa01779f904de3fc8a5b43a625146db54450c9b

**Comment by adrielparedes** [comment]
@mswiderski done!

File: kie-server-parent/kie-server-tests/pom.xml [hunk 1]
```diff
@@ -617,12 +617,13 @@
...
     </profile>    
     <profile>
+      <!-- temporary hack to let it test on jenkins id>wildfly14</id-->
       <id>wildfly11</id>
       <properties>        
         <kie.server.remoting.url>http-remoting://${container.hostname}:${container.port}</kie.server.remoting.url>
         <kie.server.context.factory>org.wildfly.naming.client.WildFlyInitialContextFactory</kie.server.context.factory>
         <org.kie.server.persistence.ds>java:/jdbc/jbpm</org.kie.server.persistence.ds>
-        <cargo.container.id>wildfly11x</cargo.container.id>
+        <cargo.container.id>wildfly14x</cargo.container.id>
```
Comment (mswiderski) [inline kie-server-parent/kie-server-tests/pom.xml]: not sure there is cargo support for wildfly 14.. already but I guess @sutaakar would know best

Comment (sutaakar) [reply kie-server-parent/kie-server-tests/pom.xml]: No, currently latest supported version is WildFly 13.

**Comment by adrielparedes** [comment]
jenkins retest this

**Comment by adrielparedes** [comment]
jenkins retest this

**Comment by adrielparedes** [comment]
jenkins retest this

**Comment by adrielparedes** [comment]
jenkins retest this

**Comment by adrielparedes** [comment]
jenkins retest this

**Comment by adrielparedes** [comment]
jenkins retest this

**Comment by adrielparedes** [comment]
jenkins retest this

**Comment by sutaakar** [comment]
There is an error in kie-server-tests module, maven-enforcer-plugin found duplicated classes.

**Comment by adrielparedes** [comment]
jenkins retest this

**Comment by mbiarnes** [comment]
Jenkins retest this please

**Comment by adrielparedes** [comment]
@ederign @sutaakar @dgutierr 
@mbiarnes created for me a Jenkins job to build all the projects because I couldn't reproduce it in my local, and finally we could run it. The run confirmed what it is just happening in pull requests Jenkins. Here is the test we made with @mbiarnes.

https://rhba-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/KIE/job/Test-jobs/job/singleRepository-Test/43/

**Comment by sutaakar** [comment]
@adrielparedes Please exclude netty-all artifact from wildfly-jms-client-bom dependency. It is an uberjar and causes class duplications with other netty dependencies.

**Comment by adrielparedes** [comment]
jenkins retest this

**Comment by adrielparedes** [comment]
jenkins retest this

**Comment by adrielparedes** [comment]
jenkins retest this

**Pull request merged**